### PR TITLE
Web API for Node search

### PR DESCRIPTION
**Pull request by dongbohu**
This PR adds a simple REST API that supports `Node` search. Here is an example:
http://34.233.249.241/api/v1/nodes/?search=abc

By default, it shows only 25 results. To get all of them, use the `limit` parameter:
http://34.233.249.241/api/v1/nodes/?search=abc&limit=100000

I also removed `frontend` test in CircleCI, because this repo is supposed to be backend only.

### CODE REVIEW

**Comment by dongbohu** [comment]
Forgot to mention: `CORS` (Cross-Origin Resource Sharing) is enabled so that anybody can `GET` from the URL `<host>/api/v1/...`. It will make the frontend development easier.

File: dj_hetmech/settings.py [hunk 1]
```diff
@@ -44,12 +44,22 @@
...
+    'rest_framework',
+    'corsheaders',
     'dj_hetmech_app',
 ]
 
+REST_FRAMEWORK = {
+    'PAGE_SIZE': 25,
+    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.LimitOffsetPagination',
+    'DEFAULT_RENDERER_CLASSES': ('rest_framework.renderers.JSONRenderer', ),
+    'DEFAULT_FILTER_BACKENDS': ('django_filters.rest_framework.DjangoFilterBackend',)
```
Comment (dhimmel) [inline dj_hetmech/settings.py]: missing comma at end of line

**Review by dhimmel — COMMENTED** [review]
Nice!

How do we know which arguments  like `search` and `limit` are available?

File: dj_hetmech/settings.py (hunk 1 above)
Comment (dongbohu) [reply dj_hetmech/settings.py]: @dhimmel Both `search` and `limit` parameters are standard in `django-rest-framework`, if you need other filters, let me know and I will add them.

**Comment by dhimmel** [comment]
Here is the output of 
```
curl --silent 'http://34.233.249.241/api/v1/nodes/?search=multiple' | python -m json.tool
```

<details>

```json
{
    "count": 7,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 14803,
            "identifier": "DOID:2377",
            "identifier_type": "str",
            "name": "multiple sclerosis",
            "url": "http://purl.obolibrary.org/obo/DOID_2377",
            "data": {
                "url": "http://purl.obolibrary.org/obo/DOID_2377",
                "source": "Disease Ontology",
                "license": "CC BY 3.0"
            },
            "metanode": "Disease"
        },
        {
            "id": 40866,
            "identifier": "C0000772",
            "identifier_type": "str",
            "name": "Multiple congenital abnormalities",
            "url": "http://identifiers.org/umls/C0000772",
            "data": {
                "url": "http://identifiers.org/umls/C0000772",
                "source": "UMLS via SIDER 4.1",
                "license": "CC BY-NC-SA 4.0"
            },
            "metanode": "Side Effect"
        },
        {
            "id": 41436,
            "identifier": "C0016655",
            "identifier_type": "str",
            "name": "Multiple fractures",
            "url": "http://identifiers.org/umls/C0016655",
            "data": {
                "url": "http://identifiers.org/umls/C0016655",
                "source": "UMLS via SIDER 4.1",
                "license": "CC BY-NC-SA 4.0"
            },
            "metanode": "Side Effect"
        },
        {
            "id": 41872,
            "identifier": "C0026769",
            "identifier_type": "str",
            "name": "Multiple sclerosis",
            "url": "http://identifiers.org/umls/C0026769",
            "data": {
                "url": "http://identifiers.org/umls/C0026769",
                "source": "UMLS via SIDER 4.1",
                "license": "CC BY-NC-SA 4.0"
            },
            "metanode": "Side Effect"
        },
        {
            "id": 42120,
            "identifier": "C0032989",
            "identifier_type": "str",
            "name": "Multiple pregnancy",
            "url": "http://identifiers.org/umls/C0032989",
            "data": {
                "url": "http://identifiers.org/umls/C0032989",
                "source": "UMLS via SIDER 4.1",
                "license": "CC BY-NC-SA 4.0"
            },
            "metanode": "Side Effect"
        },
        {
            "id": 44994,
            "identifier": "C0740281",
            "identifier_type": "str",
            "name": "Multiple allergies",
            "url": "http://identifiers.org/umls/C0740281",
            "data": {
                "url": "http://identifiers.org/umls/C0740281",
                "source": "UMLS via SIDER 4.1",
                "license": "CC BY-NC-SA 4.0"
            },
            "metanode": "Side Effect"
        },
        {
            "id": 45382,
            "identifier": "C0856120",
            "identifier_type": "str",
            "name": "Multiple sclerosis relapse",
            "url": "http://identifiers.org/umls/C0856120",
            "data": {
                "url": "http://identifiers.org/umls/C0856120",
                "source": "UMLS via SIDER 4.1",
                "license": "CC BY-NC-SA 4.0"
            },
            "metanode": "Side Effect"
        }
    ]
}
```

</details>

**Comment by dongbohu** [comment]
@dhimmel: If you have `jq` installed (`apt install jq`), you can also do:
`curl --silent 'http://34.233.249.241/api/v1/nodes/?search=multiple' | jq`
The output is nicer because it is in color.

File: dj_hetmech/settings.py (hunk 1 above)
Comment (dhimmel) [reply dj_hetmech/settings.py]: I am mostly interested in how you know which parameters are available, where that gets configured, and seeing the corresponding `django-rest-framework` documentation.

Comment (dongbohu) [reply dj_hetmech/settings.py]: I added some comments with relevant DRF documentation URL.

**Review by dhimmel — APPROVED** [review]
@dongbohu I will show you the squash merge protocol we use for merging.

**Pull request merged**